app/models/user.rb

```diff
@@ -110,4 +110,16 @@ def admin?
   def likes?(run)
     RunLike.find_by(user: self, run: run)
   end
+
+  # Retrieve all runs scoped for a paginated page excluding already loaded runs
+  def old_runs_for_page(paged_runs)
```

Comment:
This function doesn't know anything about pages, so I think its name / comment / argument should also not discuss them at all. It takes an array of runs as input and spits out an array of runs as output; doesn't matter if the input array was generated by a pagination function or not.

Reply:
I'm not sure of a good name that describes what it does that isn't a million words long.  How does `runs_for_categories` sound and then have the comment describe that it takes an array of runs and gets all runs by the user for those categories?

Reply:
Here are some ideas:
```ruby
user.previous_uploads_for(runs)
user.run_ancestors(runs)
user.runs_before(runs)

# If you're okay with changing the signature:
user.non_pbs(runs.map(&:category))
runs.ancestors # ancestors is a scope on Run
runs.previous_uploads
```
Whichever way you go yeah definitely explain what it returns in the comment, it's a complicated enough method.

Reply:
Decided to just go with the `non_pbs` and make it the inverse of `pbs`, and just group by in the view so that the method itself is fairly simple in the model.